sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -521,12 +525,37 @@ private static Exception instantiateUnexpectedException(UnexpectedExceptionInfor
 
                 switch (paramCount) {
                     case 3:
-                        return ctor.newInstance(httpRequest, responseStatusCode, responseHeaders);
+                        return Mono.just(ctor.newInstance(httpRequest, responseStatusCode, responseHeaders));
                     case 4:
-                        return ctor.newInstance(httpRequest, responseStatusCode, responseHeaders, bodyAsObject);
+                        return Mono.just(ctor.newInstance(httpRequest, responseStatusCode, responseHeaders,
+                            bodyAsObject));
                     case 5:
-                        return ctor.newInstance(httpRequest, responseStatusCode, responseHeaders, bodyAsObject,
-                            response.getDecodedHeaders().block());
+                        return response.getDecodedHeaders()
+                            .defaultIfEmpty(NullObject.INSTANCE)
```

Comment:
Can we do something like
```
return response.getDecodedHeaders()
    .map(headers -> ctor.newInstance(httpRequest, responseStatusCode,
                                        responseHeaders, bodyAsObject, headers))
    .switchIfEmpty(Mono.defer(() -> ctor.newInstance(httpRequest, responseStatusCode, responseHeaders, bodyAsObject, null)));
```

I understand what you are trying to do - but we should be able to work around the null value here.

Reply:
Done.

Reply:
With help with @anuchandy , get this piece of code use reactor way of handling null. Done.